File: Back/Gestion des conges/src/main/java/com/example/gestion_des_conges/Entities/TypeConge.java
package com.example.gestion_des_conges.Entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;
import lombok.experimental.FieldDefaults;

import javax.persistence.*;
import java.io.Serializable;
import java.util.List;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class TypeConge implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Integer idTypeConge;

    String nomType;

    String description;

    

    float nbrJours;

    @OneToMany(mappedBy = "typeConge")
    @JsonIgnore
    List<Conge> listConge;


}

File: Front/Gestion_Conge/src/app/Model/TypeConge.ts
import { Conge } from "./Conge";
import { NatureType } from "./NatureType";

export class TypeConge {
  idTypeConge: number;
  nomType: string;
  description: string;
  natureType: NatureType;
  nbrJours: number;
  listConge: Conge[];
}